proxy/http/HttpTransact.cc

```diff
@@ -6366,8 +6367,16 @@ HttpTransact::is_request_valid(State *s, HTTPHdr *incoming_request)
   RequestError_t incoming_error;
   URL *url = nullptr;
 
-  // If we are blind tunneling the header is just a synthesized placeholder anyway
+  // If we are blind tunneling the header is just a synthesized placeholder anyway.
+  // But we do have to check that we are not tunneling to a dynamic port that is
+  // not in the connect_ports list.
   if (s->client_info.port_attribute == HttpProxyPort::TRANSPORT_BLIND_TUNNEL) {
+    if (!is_port_in_range(incoming_request->url_get()->port_get(), s->http_config_param->connect_ports)) {
```

Comment:
Does this break anything else? In what circumstances would this code have executed?

Reply:
We should also check `tunnel_port_is_dynamic` actually.